Gordon360/Services/HousingService.cs

```diff
@@ -531,4 +557,228 @@ public bool ChangeApplicationDateSubmitted(int applicationID)
+                }
+                if (gender != student.Gender)
+                {
+                    throw new BadInputException() { ExceptionMessage = "The applicants are not of the same gender." };
+                }
+                gender = student.Gender;
+                var newApplicant = new Applicant
+                {
+                    ApplicationID = application_id,
+                    Applicant1 = e,
+                    Active = 1
+                }; ;
+                await context.Applicant.AddAsync(newApplicant);
+            }
+        }
+        context.SaveChanges();
+    }
+
+    /// <summary>
+    /// update the information of preferred halls
+    /// </summary>
+    /// <param name="username"> The username for the user who complete the aplication form </param>
+    /// <param name="application_id"> The ID of this application </param>
+    /// <param name="hallList"> A list of the preferred halls </param>
+    public void UpdatePreferredHall(string username, string application_id, string[] hallList)
+    {
+        var hallPreferences = hallList.Select((hall, index) => new PreferredHall { ApplicationID = application_id, Rank = index + 1, HallName = hall });
+        context.PreferredHall.AddRange(hallPreferences);
+        context.SaveChanges();
+    }
+
+    /// <summary>
+    /// update the information of preferred halls
+    /// </summary>
+    /// <param name="username"> The username for the user who complete the aplication form </param>
+    /// <param name="application_id"> The ID of this application </param>
+    /// <param name="preferenceList"> A list of the preference </param>
+    public void UpdatePreference(string username, string application_id, string[] preferenceList)
+    {
+        var preferences = preferenceList.Select((preference) => new Preference { ApplicationID = application_id, Preference1 = preference });
+        context.Preference.AddRange(preferences);
+        context.SaveChanges();
+    }
+
+    /// <summary>
+    /// update the information of due date
+    /// </summary>
+    /// <param name="dueDate"> The due date of the application </param>
+    public async Task UpdateDueDateAsync(string dueDate)
+    {
+        // This code here is not working correctly. It cannot save the changes.
+        var myDueDate = context.Config.FirstOrDefault(d => d.Key == "housing_lottery_due_date");
+        myDueDate.Value = dueDate;
+        await context.SaveChangesAsync();
+    }
+
+    /// <summary>
+    /// Remove the particular user from the current application by setting her/his active flag to 0
+    /// </summary>
+    /// <returns> Whether or not this was successful </returns>
+    public bool RemoveUser(string username)
+    {
+        var email = context.ACCOUNT.FirstOrDefault(a => a.AD_Username == username)?.email;
+        var existActiveApplicant = context.Applicant.FirstOrDefault(x => (x.Applicant1 == email) && (x.Active == 1));
```

Comment:
```suggestion
        var existingActiveApplicant = context.Applicant.FirstOrDefault(x => (x.Applicant1 == email) && (x.Active == 1));
```
Perhaps clearer wording for this var name